sematic/ui/packages/common/src/pages/RunSearch/RunList.tsx

```diff
@@ -0,0 +1,196 @@
+import RunReferenceLink from 'src/component/RunReferenceLink';
+import TableComponent from 'src/component/Table';
+import TagsList from 'src/component/TagsList';
+import RunStatusColumn from 'src/pages/RunSearch/RunStatusColumn';
+import theme from 'src/theme/new';
+
+const Container = styled.div`
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+    margin-right: -${theme.spacing(5)};
+`;
+
+const Stats = styled.div`
+    height: 50px;
+    width: 100%;
+    display: flex;
+    align-items: center;
+    flex-grow: 0;
+    flex-shrink: 0;
+`;
+
+const Pagination = styled.div`
+    height: 50px;
+    width: 100%;
+    display: flex;
+    align-items: center;
+    flex-grow: 0;
+    flex-shrink: 0;
+    justify-content: flex-end;
+    padding-right: ${theme.spacing(6)};
+
+    svg {
+        cursor: pointer;
+    }
+`;
+
+const NameSection = styled.div`
+    display: flex;
+    flex-direction: row;
+    column-gap: ${theme.spacing(2)};
+    overflow-x: hidden;
+    min-width: 100px;
+    position: relative;
+`;
+
+const StyledRunReferenceLink = styled(RunReferenceLink)`
+    color: ${theme.palette.mediumGrey.main};
+`;
+
+const RunData = {
+    ID: '3es92wd',
+    created: '2021-10-01T12:00:00Z',
+    name: 'MNIST PyTorch Example',
+    importPath: 'examples.mnist.pipeline',
+    tags: ['example', 'pytorch', 'mnist', 'demo'],
+    owner: 'Demo User',
+    status: 'RUNNING',
+    failedAt: undefined as undefined | string,
+    resolvedAt: undefined as undefined | string,
+    canceledAt: undefined as undefined | string,
+}
+
+const columnHelper = createColumnHelper<typeof RunData>()
+
+const columns = [
+    columnHelper.accessor('ID', {
+        meta: {
+            columnStyles: {
+                width: "5.923%",
+            }
+        },
+        header: 'ID',
+        cell: info => <StyledRunReferenceLink variant={'inherit'} runId={info.getValue()} />,
+    }),
+    columnHelper.accessor('created', {
+        meta: {
+            columnStyles: {
+                width: "12.3396%",
+                minWidth: "140px"
+            }
+        },
+        header: 'Submitted at',
+        cell: info => DateTimeLongConcise(info.getValue()),
+    }),
+    columnHelper.accessor(data => [data.name, data.importPath], {
+        meta: {
+            columnStyles: {
+                width: "1px",
+                maxWidth: "calc(100vw - 1090px)"
+            }
+        },
+        header: 'Name',
+        cell: info => {
+            const [name, importPath] = info.getValue();
+            return <NameSection>
+                <PipelineTitle>{name}</PipelineTitle>
+                <ImportPath>{importPath}</ImportPath>
+            </NameSection>
+        },
+    }),
+    columnHelper.accessor('tags', {
+        meta: {
+            columnStyles: {
+                width: "14.5114%",
+                minWidth: "160px"
+            }
+        },
+        header: 'Tags',
+        cell: info => <TagsList tags={info.getValue()} fold={2} />,
```

Comment:
Related to the above comment - here `fold` is an input that has to come from the outside somehow, and there is no mechanism to calculate it based on `RunData`. Please internalize its calculation inside `TagsList`.

Reply:
sounds good. Thanks for the suggestion. It is now implemented as demonstrated in [tags-list](http://localhost:6006/?path=/story/sematic-tags--tags-list)